tests/test_core.py

```diff
@@ -986,6 +986,31 @@ def test_partial_method_lots_of_args(self):
         )
         self.assertEqual(buf.value.decode("utf-8"), pystring)
 
+    def test_partial_method_arg_order(self):
+        Example = ObjCClass("Example")
+
+        self.assertEqual(Example.overloaded(0, extraArg1=0, extraArg2=0), 1)
+        self.assertEqual(Example.overloaded(0, extraArg2=0, extraArg1=0), 2)
+
+        with self.assertRaises(ValueError):
+            Example.overloaded(0, orderedArg2=0, orderedArg1=0)
```

Comment:
It would be worth adding a comment clarifying what is being tested here
```suggestion
        # Although the arguments are a unique match, they're not in the right order.
        with self.assertRaises(ValueError):
            Example.overloaded(0, orderedArg2=0, orderedArg1=0)
```